subsys/usb/device_next/class/usbd_msc_scsi.c

```diff
@@ -0,0 +1,832 @@
+/* SBC-4 Table 114 — POWER CONDITION and POWER CONDITION MODIFIER field */
+enum power_condition {
+	POWER_COND_START_VALID = 0x0,
+	POWER_COND_ACTIVE = 0x1,
+	POWER_COND_IDLE = 0x2,
+	POWER_COND_STANDBY = 0x3,
+	POWER_COND_LU_CONTROL = 0x7,
+	POWER_COND_FORCE_IDLE_0 = 0xA,
+	POWER_COND_FORCE_STANDBY_0 = 0xB,
+};
+
+SCSI_CMD_STRUCT(START_STOP_UNIT) {
+	uint8_t opcode;
+	uint8_t immed;
+	uint8_t reserved;
+	uint8_t condition;
+	uint8_t start;
+	uint8_t control;
+} __packed;
+
+#define GET_PREVENT(cmd)			(cmd->prevent & BIT_MASK(2))
+/* SBC-4 Table 77 — PREVENT field */
+enum prevent_field {
+	MEDIUM_REMOVAL_ALLOWED = 0,
+	MEDIUM_REMOVAL_SHALL_BE_PREVENTED = 1,
+	PREVENT_OBSOLETE_2 = 2,
+	PREVENT_OBSOLETE_3 = 3,
+};
+
+SCSI_CMD_STRUCT(PREVENT_ALLOW_MEDIUM_REMOVAL) {
+	uint8_t opcode;
+	uint8_t reserved1;
+	uint8_t reserved2;
+	uint8_t reserved3;
+	uint8_t prevent;
+	uint8_t control;
+} __packed;
+
+SCSI_CMD_STRUCT(READ_FORMAT_CAPACITIES) {
+	uint8_t opcode;
+	uint8_t reserved1;
+	uint8_t reserved2;
+	uint8_t reserved3;
+	uint8_t reserved4;
+	uint8_t reserved5;
+	uint8_t reserved6;
+	uint16_t allocation_length;
+	uint8_t control;
+} __packed;
+
+struct capacity_list_header {
+	uint8_t reserved1;
+	uint8_t reserved2;
+	uint8_t reserved3;
+	uint8_t capacity_list_length;
+} __packed;
+
+enum descriptor_types {
+	UNFORMATTED_OR_BLANK_MEDIA = 1,
+	FORMATTED_MEDIA = 2,
+	NO_MEDIA_PRESENT_OR_UNKNOWN_CAPACITY = 3,
+};
+
+struct current_maximum_capacity_descriptor {
+	uint32_t number_of_blocks;
+	uint8_t type;
+	uint32_t block_length : 24;
+} __packed;
+
+struct scsi_read_format_capacities_response {
+	struct capacity_list_header header;
+	struct current_maximum_capacity_descriptor desc;
+} __packed;
+
+SCSI_CMD_STRUCT(READ_CAPACITY_10) {
+	uint8_t opcode;
+	uint8_t reserved_or_obsolete[8];
+	uint8_t control;
+} __packed;
+
+struct scsi_read_capacity_10_response {
+	uint32_t last_lba;
+	uint32_t block_length;
+} __packed;
+
+SCSI_CMD_STRUCT(READ_10) {
+	uint8_t opcode;
+	uint8_t rdprotect;
+	uint32_t lba;
+	uint8_t group_number;
+	uint16_t transfer_length;
+	uint8_t control;
+} __packed;
+
+SCSI_CMD_STRUCT(WRITE_10) {
+	uint8_t opcode;
+	uint8_t wrprotect;
+	uint32_t lba;
+	uint8_t group_number;
+	uint16_t transfer_length;
+	uint8_t control;
+} __packed;
+
+static int update_disk_info(struct scsi_ctx *ctx)
+{
+	int status = disk_access_status(ctx->disk);
+
+	if (disk_access_ioctl(ctx->disk, DISK_IOCTL_GET_SECTOR_COUNT, &ctx->sector_count) != 0) {
+		ctx->sector_count = 0;
+		status = -EIO;
+	}
+	if (disk_access_ioctl(ctx->disk, DISK_IOCTL_GET_SECTOR_SIZE, &ctx->sector_size) != 0) {
+		ctx->sector_size = 0;
+		status = -EIO;
+	}
+	if (ctx->sector_size > CONFIG_USBD_MSC_SCSI_BUFFER_SIZE) {
+		status = -ENOMEM;
+	}
+	return status;
+}
+
+static size_t good(struct scsi_ctx *ctx, size_t data_in_bytes)
+{
+	ctx->status = GOOD;
+	ctx->sense_key = NO_SENSE;
+	ctx->asc = NO_ADDITIONAL_SENSE_INFORMATION;
+	return data_in_bytes;
```

Comment:
Usually we add an empty line before return
```suggestion
	ctx->asc = NO_ADDITIONAL_SENSE_INFORMATION;

	return data_in_bytes;
```